lib/openhab/dsl/rules/automation_rule.rb

```diff
@@ -159,6 +164,17 @@ def extract_event(inputs)
           struct_class.new(**inputs)
         end
 
+        #
+        # Converts inputs into context hash
+        # @return [Hash] Context hash.
+        #
+        def extract_context(inputs)
+          inputs&.reject { |key, _| key.include?(".") }
+                &.to_h do |key, value|
+                  [key.to_sym, value.is_a?(Item) ? Core::Items::Proxy.new(value) : value]
```

Comment:
This could create a difference between file-based rules and UI rules. For file-based rules, this will Proxify any item. In UI based rules, they won't be proxied, since I haven't figured out how to get to them from anywhere but the rule body.

Reply:
```ruby
return unless inputs

inputs.reject { |key, _| key.include?(".") }
      .transpose_keys(&:to_sym)
      .transpose_values { |v| v.is_a?(Item) ? Core::Items::Proxy.new(v) : v }
```

Reply:
I'm not sure what you mean here about "getting to them from anywhere but the rule body"

Reply:
Oh, I think I see what you mean now. UI Ruby scripts don't go through AutomationRule. Their context gets set directly into the JRubyScriptEngine in Java-land. So we don't get the opportunity to wrap them. I think that's acceptable for now. Proxies are mostly so that items you reference outside of rules auto-reload the correct backing item if they got removed and added. Which should be far less common in rules, since you rarely keep state beyond the rule execution. BUT... maybe we should do proxy unwrapping and re-wrapping when using shared_cache?

Reply:
With a UI Rule, the "context" is injected to the main "rule body" i.e. the plain script that you entered through the UI. 

From OpenHAB Core, when the action is executed in a UI trigger, something like:
```java
engine.put("triggeringGroup", someItem);
engine.eval(theScriptString); // <= this is the plain UI rule
```

I don't know how to access this context, i.e. the `triggeringGroup` inserted above, from inside, say, `OpenHAB::Core::Events::ItemEvent#group`.

Reply:
> maybe we should do proxy unwrapping and re-wrapping when using shared_cache?

yes, especially given the weird behaviour when storing Ruby objects into shared cache.

Reply:
`#to_h { [key, value] }` needs to stay for some reason, otherwise a lot of things would break, I don't know why.